tests/unit/lms/services/grouping_test.py

```diff
@@ -75,111 +76,85 @@ def generate_authority_provided_id(self):
 
 
 class TestUpsertWithParent:
-    def test_when_empty(self, svc):
-        course = factories.Course()
-
+    def test_it_with_empty_list(self, svc, parent_course):
         assert not svc.upsert_with_parent(
             [],
             type_=Grouping.Type.CANVAS_GROUP,
-            parent=course,
+            parent=parent_course,
         )
 
-    def test_it(self, db_session, svc):
-        # Create a parent course for the groupings to belong to.
-        # The course has to belong to svc.application_instance.
-        course = factories.Course(application_instance=svc.application_instance)
-        # A pre-existing grouping that we'll update.
-        existing_grouping = {
-            "lms_id": "existing_id",
-            "lms_name": "existing_name",
-            "extra": {"existing": "extra"},
+    def test_it_can_create_new(self, db_session, svc, parent_course):
+        attrs = {
+            "lms_id": "new_id",
+            # We'll update existing_grouping's lms_name and extra.
+            "lms_name": "new_name",
+            "extra": {"updated": "extra"},
         }
-        # Unfortunately we have to use upsert_with_parent() itself to insert the pre-existing grouping.
-        svc.upsert_with_parent(
-            [existing_grouping],
-            type_=Grouping.Type.CANVAS_GROUP,
-            parent=course,
-        )
```

Comment:
We no longer rely on the thing we are testing to test itself. We still use `generate_authority_provided_id` though, so it's not perfect.